src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathV2Spec.groovy

```diff
@@ -1303,6 +1304,106 @@ class ProtectedPathV2Spec extends HealthCheckSpecification {
     }
 
+    @Tidy
+    def "Protected path is created in different POP even if this path is not preferable"(){
+        given: "Not neighboring switch pair with three diverse paths at least"
+        def allPaths //all possible paths with 3 involved switches
+        def allPathCandidates // 3 diverse paths at least
+        def swPair = topologyHelper.getAllNotNeighboringSwitchPairs().find { swP ->
+            allPaths = swP.paths.findAll { pathHelper.getInvolvedSwitches(it).size() == 3 }
+            allPathCandidates = allPaths.unique(false) { a, b ->
+                a.intersect(b) == [] ? 1 : 0
+            } // just to avoid parallel links
+            allPaths.unique(false) { a, b ->
+                def p1 = pathHelper.getInvolvedSwitches(a)[1..-2]*.dpId
+                def p2 = pathHelper.getInvolvedSwitches(b)[1..-2]*.dpId
+                p1.intersect(p2) == [] ? 1 : 0
+            }.size() >= 3
+        } ?: assumeTrue("No suiting switches found", false)
+
+        //select paths for further manipulations
+        def mainPath1 = allPathCandidates.first()
+        def mainPath2 = allPathCandidates.find { it != mainPath1 }
+        def protectedPath = allPathCandidates.find { it != mainPath1 && it != mainPath2}
+        def involvedSwP1 = pathHelper.getInvolvedSwitches(mainPath1)*.dpId
+        def involvedSwP2 = pathHelper.getInvolvedSwitches(mainPath2)*.dpId
+        def involvedSwProtected = pathHelper.getInvolvedSwitches(protectedPath)*.dpId
+
+        and: "Src, dst and transit switches belongs to different POPs(src:1, dst:4, tr1/tr2:2, tr3:3)"
+        // tr1/tr2 for the main path and tr3 for the protected path
+        database.setSwitchPop(swPair.src.dpId, "1")
+        [involvedSwP1[1], involvedSwP2[1]].each { swId -> database.setSwitchPop(swId, "2") }
+        database.setSwitchPop(swPair.dst.dpId, "4")
+        database.setSwitchPop(involvedSwProtected[1], "3")
+
+        and: "Path which contains tr3 is non preferable"
+        /** There is not possibility to use the 'makePathNotPreferable' method,
+         * because it sets too high value for protectedPath.
+         *
+         *  totalCostOFProtectedPath should be the following:
+         *  totalCostOfMainPath + (amountOfInvolvedIsls * diversity.pop.isl.cost * diversityGroupPerPopUseCounter) - 1,
+         *  where:
+         *  diversity.pop.isl.cost = 1000
+         *  diversityGroupPerPopUseCounter = amount of switches in the same POP
+         *  (in this test there are two switches in the same POP) */
+        List<Isl> islsToUpdate = []
+        allPaths.findAll { involvedSwProtected[1] in pathHelper.getInvolvedSwitches(it)*.dpId }.each {
+            pathHelper.getInvolvedIsls(it).each {
+                islsToUpdate << it
+            }
+        }
+        def involvedIslsOfMainPath = pathHelper.getInvolvedIsls(mainPath1)
+        def totalCostOfMainPath = involvedIslsOfMainPath.sum { northbound.getLink(it).cost ?: 700 }
```

Comment:
1. done
2. I would like to avoid `assume` in the test. Actually there are only two requests(`northbound.getLink`), so I guess it is not critical to leave it as is.